### PR TITLE
Sentience fun balloon QoL

### PR DESCRIPTION
**Pull request by dragomagol**
## About The Pull Request
Makes it so that only unoccupied conscious/living mobs can be used with the sentience fun balloon.

## Why It's Good For The Game
It's no fun for a ghost to click on a prompt, expecting to play, just to find out the mob they were assigned to is already dead.

## Changelog
:cl:
qol: sentience fun balloons can no longer assign ghosts to dead mobs
qol: sentience fun balloons only work on mobs without a ckey
/:cl:

### CODE REVIEW

**Comment by zxaber** [comment]
Hey, while you're at it, can you make it skip mobs that already have a ghost? IIRC it'll replace players in the target mobs, which is not usually what I intend if I have a mix of occupied and vacant mobs.

File: code/modules/admin/fun_balloon.dm [hunk 1]
```diff
@@ -49,18 +49,18 @@
 
 /obj/effect/fun_balloon/sentience/effect()
 	var/list/bodies = list()
-	for(var/mob/living/M in range(effect_range, get_turf(src)))
-		bodies += M
+	for(var/mob/living/possessable in range(effect_range, get_turf(src)))
+		if (!possessable.ckey && possessable.stat == CONSCIOUS) // Only assign ghosts to living, non-occupied mobs!
```
Comment (ATH1909) [inline code/modules/admin/fun_balloon.dm:53]: what happens if they get given a ckey after this check but before the poll ends?

Comment (dragomagol) [reply code/modules/admin/fun_balloon.dm:53]: Based on its current functionality I would guess it'll replace the ghosts in the mobs. I can't see this being especially common though.

Comment (Cyberboss) [reply code/modules/admin/fun_balloon.dm:53]: Doesn't mean it can't be easily prevented

Comment (Mothblocks) [reply code/modules/admin/fun_balloon.dm:53]: It's an easy fix, but this is a bug that's already existed, I personally think it's not necessary for this PR.

**Pull request merged**